lib/wasm/src/module_translator.rs

```diff
@@ -94,10 +94,19 @@ pub fn translate_module<'data>(
             ParserState::BeginSection {
                 code: SectionCode::Code,
                 ..
-            } => {
-                // The code section begins
-                break;
-            }
+            } => loop {
+                match *parser.read() {
+                    ParserState::BeginFunctionBody { .. } => {}
+                    ParserState::EndSection => break,
+                    ParserState::Error(e) => return Err(WasmError::from_binary_reader_error(e)),
+                    ref s => panic!("wrong content in code section: {:?}", s),
+                }
+                let mut reader = parser.create_binary_reader();
+                let size = reader.bytes_remaining();
+                environ.define_function_body(reader
+                    .read_bytes(size)
+                    .map_err(WasmError::from_binary_reader_error)?)?;
+            },
```

Comment:
This looks good, and seeing it structured this way, it occurs to me that it'd be good to factor this out into a `parse_code_section` function, in sections_translator.rs, similar to the other `parse_*_section` functions. That way, it'd complete the separation of the module-level parsing logic from the section parsing.

(We may eventually want to get rid of `translate_module`'s top-level loop, and just parse the sections in their standard order, but that's a somewhat bigger change.)